petals-se-camel-junit/src/main/java/org/ow2/petals/camel/junit/PetalsCamelTestSupport.java

```diff
@@ -79,4 +83,23 @@ protected String getFrom(String service) {
     protected MockEndpoint getTo(String service) {
         return getMockEndpoint("petals:" + service);
     }
+
+    /**
+     * Sets the fault on the exchange's out
+     */
+    protected void setJbiFault(final MarshallingHelper marshalling, final Exchange exchange, final Object fault)
+            throws JAXBException {
+        marshalling.marshal(exchange.getOut(), fault);
+        // set this only after we are sure we properly marshaled the body!
+        setIsJbiFault(exchange);
+    }
+
+    protected void setJbiFault(final Exchange exchange, final Object fault) {
+        exchange.getOut().setBody(fault);
+        setIsJbiFault(exchange);
+    }
+
+    private void setIsJbiFault(final Exchange exchange) {
+        exchange.getOut().setHeader(PetalsCamelComponent.MESSAGE_FAULT_HEADER, true);
```

Comment:
instead of redefining those methods, can't you delegate to the static methods in `PetalsRouteBuilder` instead?